### PR TITLE
3597 allow reordering tasks of a tour

### LINKED ISSUE #3552: Improving the "tour" concept
There has just been introduced the concept of Route.

A couple remarks:

- [x] #3590
- [x] #3591
- [x] #3595
- [x] #3597
3-It is not possible to add a task to existing route folder by dragging and dropping.

### LINKED ISSUE #3597: Allow reordering tasks of a tour before assigning it
See #3552 
Needs #3595

Once the "tours" panel has been introduced, the dispatcher should be able to reorder tasks of a tour inside this panel. 
To implement this, we will need to allow PUT operations on tours on the API. 

- [x] Add a PUT operation in the [itemOperations section](https://github.com/coopcycle/coopcycle-web/blob/fdb7bc2ba02d8c371bb7602e2e63cf0b7585fe75/src/Entity/Tour.php#L21-L25) of the Tour entity. It should look like the definition of the POST operation. We will probably need to make a small change in `TourInputDataTransformer`, to make [the `name` property optional](https://github.com/coopcycle/coopcycle-web/blob/fdb7bc2ba02d8c371bb7602e2e63cf0b7585fe75/src/Api/DataTransformer/TourInputDataTransformer.php#L26). 
- [x] Add a functional test for the PUT operation in the [dispatch.feature](https://github.com/coopcycle/coopcycle-web/blob/fdb7bc2ba02d8c371bb7602e2e63cf0b7585fe75/features/dispatch.feature) file. The payload of the PUT operation should look like this: 
```json
{
  "tasks": [
     "/api/tasks/1",
     "/api/tasks/2",
     "/api/tasks/3"
  ]
}
```
- [x] As all panels share the same `DragDropContext`, everything needs to be implemented in the same [`handleDragEnd` function](https://github.com/coopcycle/coopcycle-web/blob/fdb7bc2ba02d8c371bb7602e2e63cf0b7585fe75/js/app/dashboard/redux/actions.js#L1164-L1263).

### PR DESCRIPTION
**Pull request by jimakker**
Fixes #3597
Fixes #3552
See #3595

### CODE REVIEW

**Comment by jimakker** [comment]
We have:

- third panel: unassigned Tours
- reorder tasks in unassigned Tour
- d-n-d assign tasks to unassigned Tour
- remove task from unassigned Tour
- unassign Tour from rider

Thx for reviewing, all comments welcome :heart:

File: js/shared/src/logistics/redux/actions.js [hunk 1]
```diff
@@ -4,6 +4,18 @@ export const CREATE_TASK_LIST_REQUEST = 'CREATE_TASK_LIST_REQUEST'
...
+
+export const ENABLE_UNASSIGNED_TOUR_TASKS_DROPPABLE = 'ENABLE_UNASSIGNED_TOUR_TASKS_DROPPABLE'
+export const DISABLE_UNASSIGNED_TOUR_TASKS_DROPPABLE = 'DISABLE_UNASSIGNED_TOUR_TASKS_DROPPABLE'
+
 export const createTaskListRequest = createAction(CREATE_TASK_LIST_REQUEST)
 export const createTaskListSuccess = createAction(CREATE_TASK_LIST_SUCCESS)
 export const createTaskListFailure = createAction(CREATE_TASK_LIST_FAILURE)
+
+export const enableUnassignedToursDroppable = createAction(ENABLE_UNASSIGNED_TOURS_DROPPABLE)
+export const disableUnassignedToursDroppable = createAction(DISABLE_UNASSIGNED_TOURS_DROPPABLE)
```
Comment (alexsegura) [inline js/shared/src/logistics/redux/actions.js:18]: This seems to be never called.

**Review by alexsegura — COMMENTED** [review]
Adding a task to a tour does not work. 

tours_bug

**Review by alexsegura — APPROVED** [review]
Should be ok now. 

tours_ok

**Pull request merged**